### DISCUSSION TITLE
compile fail on cygwin

### DISCUSSION THREAD

**Discussion by ghost** [aircrack-ng/aircrack-ng#2627] [General]
Hi, 
How to do to successfully compile aircrack-ng on cygwin
I hope to compile them into aircrack-ng.exe for use on Windows 10
But try to compile aircrack-ng always fail , why ? 
How should I do
I think someone should know how to do it
Thanks

```
$ cd /cygdrive/d/aircrack-ng-master

M@DESKTOP /cygdrive/d/aircrack-ng-master
$ make
make: *** No targets specified and no makefile found.  Stop.
```

```
$ cd /cygdrive/d/aircrack-ng-master/src

M@DESKTOP /cygdrive/d/aircrack-ng-master/src
$ make
make: *** No targets specified and no makefile found.  Stop.
```

﻿

**Comment by Mister-X-** [comment]
You've made errors. First, compilation should be done in the root directory, not `src/`, but you need to run a couple commands before running `make`, as mentioned in the [README](https://github.com/aircrack-ng/aircrack-ng#compiling). Follow the instructions I just mentioned.

Another detail: that installation process has been in the [wiki](https://aircrack-ng.org/doku.php?id=install_aircrack) for quite some time.

Please take the time to read the documentation before posting questions, and if possible, when there is an error, mention the steps you have taken to try to solve the issue.

**Reply by ghost** [reply]
Windows (Cygwin)
> w32api is required
> if using clang, libiconv and libiconv-devel
> make and Standard C++ Library development package

> need to run a couple commands before running , as mentioned in the [README](https://github.com/aircrack-ng/aircrack-ng#compiling).

Yeah, All equipped, 
I also read README

M@DESKTOP /cygdrive/d/aircrack-ng-master
$ ./autogen.sh
Could not auto-detect 'aclocal'; please install or specify the 'ACLOCAL' environment variable.

M@DESKTOP /cygdrive/d/aircrack-ng-master
$

Seems that installing clang on cygwin fail
No found "clang. exe"  , In "D:\cygwin64\bin\"
I can't find the reason

**Reply by Mister-X-** [reply]
You're likely missing packages. Running autoreconf -i in the root directory, here is what I get:

```
$ autoreconf -i
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, '.'.
libtoolize: copying file './ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'build/m4/stubs'.
libtoolize: copying file 'build/m4/stubs/libtool.m4'
libtoolize: copying file 'build/m4/stubs/ltoptions.m4'
libtoolize: copying file 'build/m4/stubs/ltsugar.m4'
libtoolize: copying file 'build/m4/stubs/ltversion.m4'
libtoolize: copying file 'build/m4/stubs/lt~obsolete.m4'
```

**Reply by ghost** [reply]
M@DESKTOP /cygdrive/d/aircrack-ng-master
$ autoreconf -i
-bash: autoreconf: command not found

**Reply by Mister-X-** [reply]
Exactly what I said, you are missing packages. Go through the list again and install them.
